source/detail/utils/string_helpers_impl.hpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 xlnt-community
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+//
+// @license: http://www.opensource.org/licenses/mit-license.php
+// @author: see AUTHORS file
+
+#pragma once
+
+#include "string_helpers.hpp"
+
+namespace xlnt {
+namespace detail {
+
+template<typename T>
+std::string join(const std::vector<T> &items, char delim)
```

Comment:
Hmm, are you planning to add more templated helper functions in the future? Just because the `join()` function doesn't contain many lines, so I would have just written the code in the header. But if more templated functions will be added, then I agree that it makes more sense to create an additional file. Not that it matters that much, though - I was just wondering what the reason was :slightly_smiling_face:

Reply:
> Hmm, are you planning to add more templated helper functions in the future? 

 I don't know yet.

> I was just wondering what the reason was

Just keeping the header file "clean". Maybe I was somewhat inspired by the other `*_impl.hpp` files, but I realize now that they are not used for implementation of templates. So, I will inline it (to use the same approach as used for `xlnt::optional`) and removed `string_helpers_impl.hpp`.